File: sdk/src/server-api/sc/networking/clients/ILobbyClientListener.java
package sc.networking.clients;

import sc.api.plugins.IGameState;
import sc.framework.plugins.Player;
import sc.protocol.responses.ProtocolErrorMessage;
import sc.protocol.responses.PrepareGameProtocolMessage;
import sc.protocol.responses.ProtocolMessage;
import sc.shared.GameResult;

public interface ILobbyClientListener {

  void onNewState(String roomId, IGameState state);

  void onRoomMessage(String roomId, ProtocolMessage data);

  void onError(String roomId, ProtocolErrorMessage error);

  void onGamePrepared(PrepareGameProtocolMessage response);

  void onGameLeft(String roomId);

  void onGameJoined(String roomId);

  void onGameOver(String roomId, GameResult data);

  void onGamePaused(String roomId, Player nextPlayer);

  void onGameObserved(String roomId);

}

File: sdk/src/main/server-api/sc/networking/clients/ILobbyClientListener.kt
package sc.networking.clients

import sc.api.plugins.IGameState
import sc.protocol.responses.ErrorPacket
import sc.protocol.responses.GamePreparedResponse
import sc.protocol.room.RoomMessage
import sc.shared.GameResult

// FIXME this is bound to a room anyways, can remove roomId from all signatures
/** Receives updates within a GameRoom on the client-side. */
interface ILobbyClientListener {
    fun onNewState(roomId: String, state: IGameState)
    fun onGameOver(roomId: String, data: GameResult)
    fun onRoomMessage(roomId: String, data: RoomMessage)
    fun onError(error: ErrorPacket)
    
    fun onGamePrepared(response: GamePreparedResponse)
    fun onGameJoined(roomId: String)
    fun onGameObserved(roomId: String)
}
